src/main/java/io/nats/client/impl/MessageQueue.java

```diff
@@ -34,8 +34,18 @@
     private final boolean singleThreadedReader;
     private final LinkedBlockingQueue<NatsMessage> queue;
     private final Lock filterLock;
+
+    // Poison pill is a graphic, but common term for an item that breaks loops or forces some other action
+    // In this class the poisonPill is used to break out of timed waits on the blocking queue.

```

Comment:
Consistency with PoisonPill spelling/spaces.

Reply:
@sasbury I understand what you said previously about this naming being used elsewhere, but personally I have used similar approach in the C client for a library level dispatcher and I coin the term of end message or control message. By the way, not sure if your API allows it, but I create an empty message (no subject, nothing). In your specific case, I wonder if you could even initialize it here as a static, singleton, message. The code that checks on the end message would check if this is this message (as opposed to user message). It may not be possible in Java since there is no pointer, you would have to make sure that the "equal" makes sure we know this is this message and not any user message.